### DISCUSSION TITLE
Use an external database

### DISCUSSION THREAD

**Discussion by gillesdegottex** [vercel/community#7186] [Help] [answered]
### Summary

Hi,

I'm trying to use an external postgresql database (not that of Vercel)
This database is perfectly reachable (verified using the CLI psql and a simple python script).

However, Vercel infra returns UND_ERR_CONNECT_TIMEOUT when trying to connect to this external db.

Is there any thing I should setup somewhere to allow external connection maybe?

Cheers,

### Steps to Reproduce

- Create a postregsql on, ex., fly.io: Use a static IP; Use a name ending with `-pooler` in order to match Vercel infra restrictions.
- Add the corresponding env vraiables `POSTGRES_` to the Vercel project.
- UND_ERR_CONNECT_TIMEOUT happens.

**Comment by amyegan** [comment, marked as answer]
Hey @gillesdegottex. I don't have all the error details, so I don't have a specific solution. But I do have some resources that can help you with debugging.
- [Unhandled Promise Rejections in JavaScript](https://hygraph.com/blog/unhandled-promise-rejection#unhandled-promise-rejections-in-javascript)
- [try...catch block](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch)
- [Connection Pooling with Serverless Functions](https://vercel.com/guides/connection-pooling-with-serverless-functions)

[a comment by an automated account is not shown]
